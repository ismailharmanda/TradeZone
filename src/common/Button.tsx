import {
  Text,
  StyleSheet,
  TouchableOpacity,
  ActivityIndicator,
} from 'react-native';
import {theme} from 'theme';

interface Props {
  type?: 'primary' | 'primary-outline';
  style?: any;
  onPress?: () => void;
  disabled?: boolean;
  loading?: boolean;
  text: string;
}

export const Button = ({
  type = 'primary',
  disabled,
  style,
  onPress,
  loading,
  text,
  ...props
}: Props) => {
  const buttonDisabled = disabled || loading;
  return (
    <TouchableOpacity
      disabled={buttonDisabled}
      style={[
        styles.button,
        styles[type],
        buttonDisabled && styles.disabled,
        style,
      ]}
      onPress={onPress}
      {...props}>
      {loading ? (
        <ActivityIndicator size="small" color={theme.colors.base.white} />
      ) : (
        <Text style={[styles.text]}>{text}</Text>
      )}
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    width: '100%',
    height: theme.height.button,
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: theme.radius,
  },
  primary: {
    backgroundColor: theme.colors.primary.regular,
  },
  'primary-outline': {
    backgroundColor: theme.colors.base.white,
    borderWidth: 1,
    borderColor: theme.colors.primary.regular,
  },
  disabled: {
    backgroundColor: theme.colors.primary.transparent,
  },
  text: {
    fontWeight: 'bold',
    fontSize: theme.text.size.md.fontSize,
    lineHeight: theme.text.size.md.lineHeight,
    color: theme.colors.secondary.regular,
  },
});
