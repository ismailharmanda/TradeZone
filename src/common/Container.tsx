import React from 'react';
import {Image, StyleSheet, KeyboardAvoidingView, ViewStyle} from 'react-native';
import {theme} from 'theme';

interface Props {
  children: React.ReactNode;
  withLogo?: boolean;
  containerStyle?: ViewStyle;
}

export const Container = ({
  children,
  withLogo = true,
  containerStyle,
}: Props) => (
  <KeyboardAvoidingView
    style={[styles.container, containerStyle]}
    behavior="padding">
    <Image
      source={require('../assets/logo.png')}
      resizeMode="contain"
      style={[styles.logo, withLogo ? null : styles.hide]}
    />
    {children}
  </KeyboardAvoidingView>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: '100%',
    paddingHorizontal: theme.spacing.screenHorizontalPadding,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: theme.colors.base.white,
    gap: theme.spacing.md,
  },
  logo: {
    width: '40%',
    height: '20%',
  },
  hide: {
    display: 'none',
  },
});
