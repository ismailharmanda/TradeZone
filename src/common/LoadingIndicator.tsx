import {
  View,
  ActivityIndicator,
  Text,
  StyleSheet,
  Dimensions,
} from 'react-native';
import {theme} from 'theme';

type Props = {
  isLoading: boolean;
  text?: string;
};

export const LoadingIndicator = ({isLoading, text = 'Loading...'}: Props) => {
  return (
    <View
      testID="loading-indicator"
      style={[styles.container, !isLoading && styles.hide]}>
      <ActivityIndicator
        size={'large'}
        style={styles.activityIndicator}
        color={theme.colors.base.white}
      />
      <Text style={styles.text}>{text}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    justifyContent: 'flex-start',
    alignItems: 'center',
    alignSelf: 'center',
    height: Dimensions.get('window').height,
    width: Dimensions.get('window').width,
    backgroundColor: 'rgba(0, 0,0, 0.1)',
    zIndex: 2,
  },
  activityIndicator: {
    transform: [{scaleX: 2}, {scaleY: 2}],
    top: '25%',
  },
  hide: {
    display: 'none',
  },
  text: {
    textAlign: 'center',
    color: theme.colors.base.white,
    fontSize: theme.text.size.lg.fontSize,
    lineHeight: theme.text.size.lg.lineHeight,
    fontWeight: 'bold',
    top: '30%',
    marginHorizontal: 30,
  },
});
