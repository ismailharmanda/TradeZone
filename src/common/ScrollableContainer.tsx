import React from 'react';
import {StyleSheet, ScrollView, ViewStyle} from 'react-native';

import {theme} from 'theme';

interface Props {
  children: React.ReactNode;
  containerStyle?: ViewStyle;
  contentContainerStyle?: ViewStyle;
}

export const ScrollableContentContainer = ({
  children,
  containerStyle,
  contentContainerStyle,
}: Props) => (
  <ScrollView
    testID="scrollableContentContainer"
    contentContainerStyle={[styles.contentContainer, contentContainerStyle]}
    style={[styles.container, containerStyle]}>
    {children}
  </ScrollView>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: '100%',
    paddingHorizontal: theme.spacing.screenHorizontalPadding,
    backgroundColor: theme.colors.base.white,
    gap: theme.spacing.md,
  },
  contentContainer: {
    flexGrow: 1,
    paddingVertical: theme.spacing.screenVerticalPadding,
    justifyContent: 'flex-start',
    alignItems: 'flex-start',
  },
});
