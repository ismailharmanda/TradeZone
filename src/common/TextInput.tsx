import React from 'react';
import {
  View,
  StyleSheet,
  Text,
  TextInput as Input,
  ViewStyle,
} from 'react-native';
import {theme} from 'theme';
import {TextInputProps} from 'react-native';

interface Props {
  containerStyle?: ViewStyle;
  placeholder?: string;
  disabled?: boolean;
  loading?: boolean;
  errorText?: string;
  value?: string;
  onChangeText?: (text: string) => void;
  textContentType?: TextInputProps['textContentType'];
  error?: boolean;
}

export const TextInput = ({
  containerStyle,
  value,
  disabled,
  onChangeText,
  errorText,
  textContentType,
  placeholder,
  ...props
}: Props) => (
  <View style={[styles.container, containerStyle]}>
    <Input
      autoCapitalize="none"
      keyboardType={
        textContentType === 'emailAddress' ? 'email-address' : 'default'
      }
      secureTextEntry={textContentType === 'password'}
      textContentType={textContentType}
      pointerEvents={disabled ? 'none' : 'auto'}
      value={value}
      onChangeText={onChangeText}
      style={styles.input}
      placeholder={placeholder}
      {...props}
    />
    {!!errorText && <Text style={styles.error}>{errorText}</Text>}
  </View>
);

const styles = StyleSheet.create({
  container: {
    width: '100%',
  },
  input: {
    backgroundColor: theme.colors.secondary.regular,
    color: theme.colors.base.dark,
    height: theme.height.input,
    borderRadius: theme.radius,
    paddingHorizontal: theme.spacing.sm,
    fontWeight: 'bold',
  },
  error: {
    fontSize: theme.text.size.sm.fontSize,
    lineHeight: theme.text.size.sm.lineHeight,
    color: theme.colors.error.regular,
    paddingHorizontal: theme.spacing.xs,
    paddingTop: theme.spacing.xs,
  },
});
