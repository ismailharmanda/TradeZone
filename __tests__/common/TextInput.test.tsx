import React from 'react';
import {render, fireEvent} from 'react-native-testing-library';
import renderer from 'react-test-renderer';
import {TextInput} from 'common/TextInput';

describe('TextInput', () => {
  const onChangeText = jest.fn();
  const placeholder = 'Enter text';
  const value = 'Hello';
  const errorText = 'Invalid input';

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <TextInput
          value={value}
          onChangeText={onChangeText}
          placeholder={placeholder}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the component correctly', () => {
    const {getByPlaceholder} = render(
      <TextInput
        value={value}
        onChangeText={onChangeText}
        placeholder={placeholder}
      />,
    );

    const input = getByPlaceholder(placeholder);
    expect(input.props.value).toBe(value);
  });

  it('calls onChangeText when the input text changes', () => {
    const {getByPlaceholder} = render(
      <TextInput
        value={value}
        onChangeText={onChangeText}
        placeholder={placeholder}
      />,
    );

    const input = getByPlaceholder(placeholder);
    fireEvent.changeText(input, 'New text');

    expect(onChangeText).toHaveBeenCalledWith('New text');
  });

  it('displays error text when error is true', () => {
    const {getByText} = render(
      <TextInput
        value={value}
        onChangeText={onChangeText}
        placeholder={placeholder}
        errorText={errorText}
        error
      />,
    );

    const error = getByText(errorText);
    expect(error).toBeDefined();
  });
});
