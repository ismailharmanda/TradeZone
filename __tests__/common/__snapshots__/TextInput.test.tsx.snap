// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders correctly 1`] = `
<View
  style={
    [
      {
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <TextInput
    autoCapitalize="none"
    keyboardType="default"
    onChangeText={[MockFunction]}
    placeholder="Enter text"
    pointerEvents="auto"
    secureTextEntry={false}
    style={
      {
        "backgroundColor": "rgb(240,237,246)",
        "borderRadius": 8,
        "color": "rgb(255,119,5)",
        "fontWeight": "500",
        "height": 48,
        "paddingHorizontal": 8,
      }
    }
    value="Hello"
  />
</View>
`;
